pkg/common/common.go

```diff
@@ -78,4 +80,27 @@ const (
 
 	// DefaultResyncPeriod sets a 10 minute resync period, used in the controller pkg and the controller cmd executable
 	DefaultResyncPeriod = 10 * time.Minute
+
```

Comment:
Thanks for adding these, I don't know if we need the full blown mapping or not; for example I had just added a new member to the base controller "ctrlType: string" and variables (not constants even though we LOVE constants) for importer, cloner, uploader.  That way the NewController function would just expect that to be set, and it's super easy to access (c.ctrlType).

The only thing I don't like about your approach is using this common.go file (it can become a dumping ground again rather quickly) and exporting everything.  Not a big deal at all but just something to throw out there if you're interested in considering it.